src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -115,6 +116,7 @@ public ImageReferenceTypeClass(
                 String galleryName,
                 String galleryImageDefinition,
                 String galleryImageVersion,
+                boolean galleryImageSpecialized,
```

Comment:
can you try add this via a setter instead?

Generally should avoid adding non mandatory attributes in the constructor for Jenkins data binding.

The data structure here is not great already and could do with some cleanup though

Reply:
Added DataBoundSetter for this parameter.